### PR TITLE
DAT-3870 | Add entry point to source editor

### PR DESCRIPTION
**Pull request by rogatty**
## Links

* https://wikia-inc.atlassian.net/browse/DAT-3870

## Description

Allow user to open source editor from Infobox Builder.

## Reviewers

@RafLeszczynski

### CODE REVIEW

File: front/main/app/templates/components/infobox-builder-sidebar.hbs [hunk 1]
```diff
@@ -19,3 +19,5 @@
 		{{/each}}
 	</div>
 {{/if}}
+
+<a {{action attrs.goToSourceEditor}}>[ ] MAKE ME PRETTY</a>
```
Comment (RafLeszczynski) [inline front/main/app/templates/components/infobox-builder-sidebar.hbs]: please make it pretty :-)

File: front/main/app/routes/infobox-builder.js [hunk 2]
```diff
@@ -114,6 +113,23 @@ export default Ember.Route.extend(ConfirmationMixin, {
...
+			const ponto = window.Ponto;
+
+			ponto.invoke(
+				'wikia.infoboxBuilder.ponto',
+				'redirectToSourceEditor',
+				null,
+				() => {
+
+				},
+				function (data) {
```
Comment (RafLeszczynski) [inline front/main/app/routes/infobox-builder.js]: could you use arrow function here?

File: front/main/app/routes/infobox-builder.js [hunk 3]
```diff
@@ -95,17 +95,16 @@ export default Ember.Route.extend(ConfirmationMixin, {
...
-		redirectToTemplatePage(title) {
+		redirectToTemplatePage() {
 			return new Ember.RSVP.Promise((resolve, reject) => {
 				const ponto = window.Ponto;
 
 				ponto.invoke(
 					'wikia.infoboxBuilder.ponto',
 					'redirectToTemplatePage',
-					title,
+					null,
```
Comment (RafLeszczynski) [inline front/main/app/routes/infobox-builder.js]: why `null` instead of `title`

File: front/main/app/controllers/infobox-builder.js [hunk 4]
```diff
@@ -6,35 +6,43 @@ export default Ember.Controller.extend({
...
 		/**
-		 * @desc Saves infobox state to template and calls redirect method on route.
-		 * on model and connect with <iframe> parent to redirect to another page.
+		 * Saves infobox state to template and calls redirect method on route.
+		 *
+		 * @param {Boolean} [redirectToTemplatePage=true]
 		 * @returns {Ember.RSVP.Promise}
 		 */
-		save() {
+		save(redirectToTemplatePage = true) {
```
Comment (RafLeszczynski) [inline front/main/app/controllers/infobox-builder.js]: could you use different name than `redirectToTemplatePage` as it's already use for the action name

File: front/main/app/components/infobox-builder.js [hunk 5]
```diff
@@ -146,6 +131,25 @@ export default Ember.Component.extend(
 				this.get('cancelAction')();
 			},
 
+			goToSourceEditor() {
+				const controllerAction = this.get('goToSourceEditor');
+
+				if (this.get('isDirty') && window.confirm('Wanna save?')) {
```
Comment (RafLeszczynski) [inline front/main/app/components/infobox-builder.js]: shouldn't this message be translatable?

File: front/main/app/components/infobox-builder.js [hunk 6]
```diff
@@ -146,6 +131,25 @@ export default Ember.Component.extend(
 				this.get('cancelAction')();
 			},
 
+			goToSourceEditor() {
```
Comment (RafLeszczynski) [inline front/main/app/components/infobox-builder.js]: what do you think about some test for this method?

File: front/main/app/routes/infobox-builder.js (hunk 3 above)
Comment (rogatty) [inline front/main/app/routes/infobox-builder.js]: It's a param that goes to:
```
this.redirectToTemplatePage = function () {
	w.location = w.templatePageUrl;
	return true;
};
```
which doesn't accept any params. I guess it was different before and this is just a clean up.

File: front/main/app/styles/component/_infobox-builder.scss [hunk 7]
```diff
@@ -133,36 +149,31 @@ $spinner-text-color: rgba(0, 0, 0, .5);
...
+		&:hover .icon-button-icon {
 			fill: $color-gray-dark;
 		}
 	}
 }
 
+
+
+
+
```
Comment (dianafa) [inline front/main/app/styles/component/_infobox-builder.scss]: what happened here?

File: front/main/app/routes/infobox-builder.js [hunk 8]
```diff
@@ -114,6 +113,26 @@ export default Ember.Route.extend(ConfirmationMixin, {
...
+		},
+
+		/**
+		 * Connects with ponto and redirects to source editor
+		 *
+		 * @returns {void}
+		 */
+		goToSourceEditor() {
+			const ponto = window.Ponto;
+
```
Comment (dianafa) [inline front/main/app/routes/infobox-builder.js]: would it make sense to add check `(ponto && typeof ponto.invoke === 'function')` here?

File: front/main/app/styles/component/_infobox-builder.scss (hunk 7 above)
Comment (rogatty) [inline front/main/app/styles/component/_infobox-builder.scss]: Enter gone rogue.

**Comment by dianafa** [comment]
one small question, besides it LGTM

File: front/main/app/routes/infobox-builder.js (hunk 8 above)
Comment (RafLeszczynski) [inline front/main/app/routes/infobox-builder.js]: if `ponto` load fails we want be able to call this function so I'd skip the check

**Comment by RafLeszczynski** [comment]
waiting for the tests to pass before moving to QA

File: front/main/app/templates/components/infobox-builder.hbs [hunk 9]
```diff
@@ -64,6 +64,14 @@
 			ns='infobox-builder'}}</p>
 		{{/pop-over}}
 	</div>
+	{{#modal-dialog isVisible=showGoToSourceModal class='infobox-builder-go-to-source-modal'}}
+		{{infobox-builder-go-to-source-modal
+			message=(i18n 'main.source-editor-save-confirmation' ns='infobox-builder')
+			goToSource=(action 'goToSource')
+			goToSourceButtonLabel='No'
+			saveAndGoToSourceButtonLabel='Yes'
```
Comment (dianafa) [inline front/main/app/templates/components/infobox-builder.hbs]: shouldn't this labels be translatable?

File: front/main/app/components/infobox-builder.js [hunk 10]
```diff
@@ -128,30 +129,49 @@ export default Ember.Component.extend(
...
+					this.handleGoToSource();
+				}
+			},
+
+			/**
+			 * @param {Boolean} saveChanges
+			 * @returns {void}
+			 */
+			goToSource(saveChanges) {
+				const trackingLabel = `go-to-source-modal-${saveChanges ? 'save-changes-and-' : ''}go-to-source`;
```
Comment (Warkot) [inline front/main/app/components/infobox-builder.js]: Concatenating message keys looks fancy in the code, but grepping that is troublesome.

File: front/main/app/templates/components/infobox-builder.hbs [hunk 11]
```diff
@@ -52,7 +52,22 @@
...
+			id='go-to-source'
+		}}
+		{{#pop-over for="go-to-source" on="hover" flow='leftCenter'}}
+			<p class="infobox-builder-go-to-source-tooltip-text">{{i18n 'main.source-editor-tooltip'
+			ns='infobox-builder'}}</p>
+		{{/pop-over}}
 	</div>
+	{{#modal-dialog isVisible=showGoToSourceModal class='infobox-builder-go-to-source-modal'}}
+		{{infobox-builder-go-to-source-modal goToSource=(action 'goToSource')
+		}}
```
Comment (Warkot) [inline front/main/app/templates/components/infobox-builder.hbs]: Why the `\n`? OR why no `\n` after component name?

**Comment by dianafa** [comment]
LGTM :dancers:

**Pull request merged**